### PR TITLE
Various changes

### PR DESCRIPTION
**Pull request by 7PH**
- Support HTTP_ proxy env variables
- Add support for sonar.scanner.app, sonar.log.level and sonar.verbose
- Fix scanner-engine parser not being able to parse multi-lines JSON buffers
- Support forwarding proxy conf to scanner-engine
- Fix JRE provisioning to fallback on native java & Fix OS detection to support aliases
- Drop +x permission hotfix
- Drop cache cleanup methods
- Reset package-lock.json to drop repox entries

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -106,14 +113,14 @@ export function defineSonarScannerParams(
...
-  if (params.token) {
-    sonarScannerParams['sonar.token'] = params.token;
+  if (scanOptions.token) {
+    sonarScannerParams['sonar.token'] = scanOptions.token;
   }
-  if (params.options) {
-    sonarScannerParams = Object.assign(sonarScannerParams, params.options);
+  if (scanOptions.options) {
+    sonarScannerParams = Object.assign(sonarScannerParams, scanOptions.options);
   }
```
Comment (lucas-paulger-sonarsource) [inline src/config.ts:124]: might have overlapped a bit on the params, so we can sync and make sure we get the version we want merged

File: src/java.ts [hunk 2]
```diff
@@ -43,15 +46,15 @@ function supportsJreProvisioning(serverUrl: string, serverVersion: SemVer) {
...
   platformInfo: PlatformInfo,
+  scanOptions: ScanOptions,
 ): Promise<
   JreMetaData & {
     jrePath: string;
   }
 > {
   const { data } = await axios.get<JreMetaData>(
-    `${serverUrl}/api/v2/scanner/jre/info?os=${platformInfo.os}&arch=${platformInfo.arch}`,
+    `${scanOptions.serverUrl}/api/v2/scanner/jre/info?os=${platformInfo.os}&arch=${platformInfo.arch}`,
```
Comment (lucas-paulger-sonarsource) [inline src/java.ts:57]: we will conflict here, but the new API routes need to be used

File: src/platform.ts [hunk 3]
```diff
@@ -27,11 +25,21 @@ function isAlpineLinux(): boolean {
   return !!content && (content.match(/^ID=([^\u001b\r\n]*)/m) || [])[1] === 'alpine';
 }
 
-export function getPlatformInfo(): PlatformInfo {
-  const os = isAlpineLinux() ? 'alpine' : process.platform;
+function getSupportedOS(): SupportedOS {
+  const mapping: { [nodePlatform: string]: SupportedOS } = {
+    linux: isAlpineLinux() ? 'alpine' : 'linux',
+    // TODO: Check whether these aliases are necessary
```
Comment (lucas-paulger-sonarsource) [inline src/platform.ts:31]: for the POC I'm sure they aren't but we can create a ticket to ask this at the right time

File: tsconfig.json [hunk 4]
```diff
@@ -39,7 +39,7 @@
     // "resolvePackageJsonExports": true,                /* Use the package.json 'exports' field when resolving package imports. */
     // "resolvePackageJsonImports": true,                /* Use the package.json 'imports' field when resolving imports. */
     // "customConditions": [],                           /* Conditions to set in addition to the resolver-specific defaults when resolving imports. */
-    // "resolveJsonModule": true,                        /* Enable importing .json files. */
+    "resolveJsonModule": true,                        /* Enable importing .json files. */
```
Comment (lucas-paulger-sonarsource) [inline tsconfig.json:42]: what file do we evaluate for json?

**Review by lucas-paulger-sonarsource — APPROVED** [review]

File: tsconfig.json (hunk 4 above)
Comment (7PH) [reply tsconfig.json:42]: the `package.json` [here](https://github.com/7PH/sonar-scanner-npm-mmf-3712-poc/pull/3/files#diff-c3095d5010e65c52737a98a5d618ea24049ebe90c8470752426081d70ed6e012R23)

File: src/platform.ts (hunk 3 above)
Comment (7PH) [reply src/platform.ts:31]: agreed

**Pull request merged**